socketioxide/src/payload_value/binary.rs

```diff
@@ -0,0 +1,80 @@
+use std::ops::{Deref, DerefMut};
+
+use serde::de;
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct Binary(Vec<u8>);
```

Comment:
`serde` has a `Serialize` impl for `Vec<T>` that always serializes as an array of numbers.  Unfortunately they don't have a special version for `Vec<u8>` that instead calls `serialize_bytes()`.  To combat this, I added a `Binary(Vec<u8>)` wrapper type that implements `Serialize` in the way we need.  This is of course a little ergonomically painful.  I noticed that `bytes::Bytes` actually does implement things properly.  Would it make sense to drop the `Binary` wrapper and require people use `Bytes` instead (which we can re-export from socketioxide to make things easier)?

Reply:
Make sense!  I've started integrating `Bytes`, and I'm realizing that it's silly to use `Vec<u8>` for the `PayloadValue::Binary()` variant; copying back and forth between the vec and `Bytes` is wasteful.  But after changing that, I realized it makes sense for `Bytes` to be used in `BinaryPacket` too.  That's as far as I'll go, though: if you think it makes sense for `engineioxide` to use `Bytes` as well (it has some uses for it already, but there are still a bunch of uses of `Vec<u8>`), that's probably for a separate PR.